src/libs/ReportUtils.js

```diff
@@ -432,6 +432,27 @@ function getPolicyName(report) {
     return policy.name || report.oldPolicyName || Localize.translateLocal('workspace.common.unavailable');
 }
 
+/**
+ * Checks if the current user is allowed to comment on the given report.
+ * @param {Object} report
+ * @param {String} report.writeCapability
+ * @param {Object} policy
+ * @param {String} policy.role
+ * @returns {Boolean}
+ */
+function isAllowedToComment(report, policy) {
+    // Default to allowing all users to post
+    const capability = lodashGet(report, 'writeCapability', CONST.REPORT.WRITE_CAPABILITIES.ALL);
```

Comment:
> How does this work? I don't have access to the PR.

Sorry about the video! So when the `writeCapability` is set, we push the Onyx updates to any accounts on the report. And in the `OpenReport` API call we are returning the value of `writeCapability`, defaulting to `null` when unset.

If we default to `all` instead of `null` in the API response, does that equally resolve the concern you've outlined? (cc @Beamanator since we were debating the default in the Web PR initially too)

Reply:
Okay, point 1 implemented in https://github.com/Expensify/Web-Expensify/pull/37527 and point 2 added to this PR!